data/dataset.py

```diff
@@ -0,0 +1,97 @@
+import random
+
+import torch
+import numpy as np
+from torch.utils.data import Dataset
+from datasets import Dataset as HFDataset
+
+from data.masking import AwesomeMasks
+from data.tokenizer import MidiEncoder
+from data.quantizer import MidiQuantizer
+from data.augmentation import pitch_shift, change_speed
+
+
+class MidiDataset(Dataset):
+    def __init__(
+        self, dataset: HFDataset, tokenizer: MidiEncoder, augmentation_probability: float = 0.0, masking_probability: float = 0.15
+    ):
+        super().__init__()
+
+        self.dataset = dataset
+        self.quantizer = MidiQuantizer(7, 7, 7)
+        self.tokenizer = tokenizer
+        self.masks = AwesomeMasks()
+
+        self.augmentation_probability = augmentation_probability
+        self.masking_probability = masking_probability
+
+    def __len__(self):
+        return len(self.dataset)
+
+    def apply_augmentation(self, record: dict):
+        # shift pitch augmentation
+        if random.random() < self.augmentation_probability:
+            shift = 7
+            record["pitch"] = pitch_shift(record["pitch"], shift)
```

Comment:
```suggestion
            shift_threshold = 7
            record["pitch"] = pitch_shift(
                pitch=record["pitch"],
                shift_threshold=shift_threshold,
            )
```

Please try to use named arguments as much as possible - it's a big difference here between shift and shift threshold.

I'm also thinking about our ability to track exactly what kind of augmentation the went through, so we may want to conisder tracking the actual shift value here :thinking: